NRF Cloud MSS: native sim support (#22155)
nrfconnect/sdk-nrf: samples/cellular/nrf_cloud_multi_service/README.rst

If you were directed here as part of other instructions, proceed to the next step of those instructions.

> Running on native_sim

Reviewer: This section should be part of Building and running section:
https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/samples/cellular/nrf_cloud_multi_service/README.html#building_and_running

Also I would recommend renaming the title for "Building with native_sim"

Reviewer 2: Otherwise, good suggestion, but can we open the native_sim?

=====================

> This sample can be run on the `native_sim` target. That way, you can try out connectivity without the need for embedded hardware.
> A Linux host or docker container is required to run the `native_sim` target.
> Some setup is needed in order to connect the application to the network. See the :ref:`networking_with_native_sim` for more information. Here, the `nat.conf` configuration is used.
> Refer to :ref:`nrf_cloud_multi_service_create_device_cred_locally` for creating the necessary device credentials locally.
> Then, modify the `overlay_native_sim.conf` file to use the client id for the credentials you created.

Reviewer 2: ```suggestion
You can run this sample on the :ref:`zephyr:native_sim` target.
This enables you to try out connectivity without the need for embedded hardware.
A Linux host or docker container is required to run the ``native_sim`` target.
Some setup is needed to connect the application to the network.
See :ref:`zephyr:networking_with_native_sim` for more information.
This sample uses the :file:`nat.conf` configuration.
Refer to :ref:`nrf_cloud_multi_service_create_device_cred_locally` for creating the necessary device credentials locally.
Modify the :file:`overlay_native_sim.conf` file to use the client ID for the credentials you created.
```

Reviewer 3: I agree with Pekka's comments and have no further comments of my own. I'm okay to approve once his questions are answered and suggestions are applied.

   .. code-block:: console

      # build nrf_cloud_multi_service with native_sim configs, optionally use -DEXTRA_CONF_FILE="overlay_native_sim.conf;overlay_native_sim_coap.conf" for CoAP
      west build -b native_sim -- -DEXTRA_CONF_FILE="overlay_native_sim.conf" -DSB_CONF_FILE="sysbuild_native_sim.conf"
      # set up zephyr ethernet interface
      ~/work/ncs/tools/net-tools$ sudo ./net-setup.sh --config nat.conf start
      # run the application (use CTRL-C to exit)
      ./build/nrf_cloud_multi_service/zephyr/zephyr.exe
      # clean up zephyr ethernet interface
      ~/work/ncs/tools/net-tools$ sudo ./net-setup.sh --config nat.conf stop

> If you intend to trace network traffic on the `zeth` interface, it can be useful to modify the `ssl_tls12_populate_transform` function inside `mbedtls` close to `MBEDTLS_SSL_KEY_EXPORT_TLS12_MASTER_SECRET` to print the TLS ephemeral keys:

Reviewer 2: ```suggestion
If you intend to trace network traffic on the ``zeth`` interface, it can be useful to modify the :c:func:`ssl_tls12_populate_transform` function inside ``mbedtls`` close to ``MBEDTLS_SSL_KEY_EXPORT_TLS12_MASTER_SECRET`` to print the TLS ephemeral keys:
```

   .. code-block:: c

      printf("\nCLIENT_RANDOM ");

      for (size_t i = 0; i < 32; i++) {
         printf("%02x", randbytes[32+i]);
      }

      printf(" ");

      for (size_t i = 0; i < 48; i++) {
         printf("%02x", master[i]);
      }

      printf("\n\n");

> This will allow you to decrypt the traffic using Wireshark. Copy this line into a `dtls_keys.log` file and use the following command to decrypt the traffic:

Reviewer 2: ```suggestion
This allows you to decrypt the traffic using Wireshark.
Copy this line into the :file:`dtls_keys.log` file and use the following command to decrypt the traffic:
```
Which line one would need to copy?
